fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java

```diff
@@ -108,6 +109,20 @@ public String toSearchSelfUri() throws URISyntaxException {
 
     }
 
+    private void appendResourceTypesParameter() {
+        if (context.getSearchResourceTypes() != null && !context.getSearchResourceTypes().isEmpty()) {
+            queryString.append(SearchConstants.AND_CHAR);
+            queryString.append(SearchConstants.RESTYPE);
+            queryString.append(SearchConstants.EQUALS_CHAR);
+            String delim = SearchConstants.EMPTY_QUERY_STRING;
+            for (String param : context.getSearchResourceTypes()) {
+                queryString.append(delim + param);
```

Comment:
break into two appends 

append the DELIMITER char and then the param string 
it'll be slightly more efficient. 
you won't create a string, and then append the created string. 
you'll add a char, then a string.

Reply:
Also you could use StringJoiner here.

Reply:
make sense to use append instead of + in this cases, will change

Reply:
done